fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerLifecycleEvents.java

```diff
@@ -116,6 +129,41 @@ public interface ServerStopped {
 		void onServerStopped(MinecraftServer server);
 	}
 
+	@FunctionalInterface
+	public interface SyncDataPackContents {
+		/**
+		 * Called right before tags and recipes are sent to one or multiple clients.
+		 * The {@linkplain MinecraftServer#getResourceManager() server resource manager} is up-to-date when this is called.
+		 *
+		 * <p>If {@code player} is non-{@code null}, then that player just joined and data should only be sent to that player.
+		 * Otherwise, if {@code player} is {@code null}, the server just finished a reload and data should be sent to all players in the {@code playerManager}.
+		 *
+		 * <p>Usage example, to send a packet to each relevant player with custom data:
+		 * <pre>{@code
+		 * ServerLifecycleEvents.SYNC_DATA_PACK_CONTENTS.register((playerManager, player) -> {
+		 *     // 1) Encode packet
+		 *     PacketByteBuf buf = PacketByteBufs.create();
+		 *     // Fill buffer with data...
+		 *     // Create packet
+		 *     Packet<?> packet = ServerPlayNetworking.createS2CPacket(CHANNEL_NAME, buf);
+		 *
+		 *     // 2) Send to relevant player(s)
+		 *     if (player == null) {
+		 *         // Reload: send to all
+		 *         playerManager.sendToAll(packet);
+		 *     } else {
+		 *         // Player joined: send only to that player
+		 *         player.networkHandler.sendPacket(packet);
+		 *     }
```

Comment:
This feels weird, like there's two events in one.

Reply:
Alternatives:
- Having two events.
- Sending the event for each player, and needing to create a new packet every time.

I don't think the alternatives are particularly nicer than this, especially with the extensive javadoc explaining how to use the event.

Reply:
Changed to sending the event for each player, the API shape is a lot better, and I think trying to not serialize the packet multiple times is premature optimization. If this proves to be a bottleneck, we can always introduce a specialized event later on.